Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+    case missingArgument(String)
+    case noCollectionMatchingURL(String)
+}
+
+extension CollectionsError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .invalidVersionString(let versionString):
+            return "invalid version string '\(versionString)'"
+        case .missingArgument(let argumentName):
+            return "missing argument '\(argumentName)'"
+        case .noCollectionMatchingURL(let url):
+            return "no collection matching URL '\(url)'"
+        }
+    }
+}
+
+struct JSONOptions: ParsableArguments {
+    @Flag(name: .long, help: "Output as JSON")
+    var json: Bool = false
+}
+
+struct ProfileOptions: ParsableArguments {
+    @Option(name: .long, help: "Profile to use for the given command")
+    var profile: String?
+
+    var usedProfile: PackageCollectionsModel.Profile {
+        if let profile = profile {
+            return .init(name: profile)
+        } else {
+            return .default
+        }
+    }
+}
+
+public struct SwiftPackageCollectionsTool: ParsableCommand {
+    public static var configuration = CommandConfiguration(
+        commandName: "package-collections",
+        _superCommandName: "swift",
+        abstract: "Interact with package collections",
+        discussion: "SEE ALSO: swift build, swift package, swift run, swift test",
+        version: Versioning.currentVersion.completeDisplayString,
+        subcommands: [
+            Add.self,
+            Describe.self,
+            DescribeCollection.self,
+            List.self,
+            ProfileList.self,
+            Refresh.self,
+            Remove.self,
+            Search.self
+        ],
+        helpNames: [.short, .long, .customLong("help", withSingleDash: true)])
+
+    public init() {
+    }
+
+    // MARK: Profiles
+
+    struct ProfileList: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "List configured profiles")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        mutating func run() throws {
+            let profiles: [PackageCollectionsModel.Profile] = try await { self.collections.listProfiles(callback: $0) }
+
+            if jsonOptions.json {
+                try JSONEncoder().print(profiles)
+            } else {
+                profiles.forEach {
+                    print($0)
+                }
+            }
+        }
+    }
+
+    // MARK: Collections
+
+    struct List: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "List configured collections")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            let collections = try await { self.collections.listCollections(identifiers: nil, in: profileOptions.usedProfile, callback: $0) }
+
+            if jsonOptions.json {
+                try JSONEncoder().print(collections)
```

Comment:
Need `withoutEscapingSlashes` I think

Reply:
We'll get it through changing the implementation of `print` itself, as you mentioned above.